### DISCUSSION TITLE
quasar build with dev env (without running on local server)

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#11010] [General - Components / Directives / etc]
I'd like to build my quasar project in dev env (without running on local server) from the terminal (and eventually in package.json).
i.e. an actual build, same output as "quasar build", except with NODE_ENV set to development (and everything that stems from it).

I need it for my staging environment.

I tried the following, nothing worked so far:

```
quasar build --QENV development
quasar build --env.QENV development
QENV=development quasar build

quasar build --NODE_ENV development
quasar build --env.NODE_ENV development
NODE_ENV=development quasar build
```

**Comment by MilosPaunovic** [comment]
You could add [cross-env](https://www.npmjs.com/package/cross-env) package and run:

```bash
cross-env NODE_ENV=development quasar build
```

**Reply by ghost** [reply]
@MilosPaunovic Just tried it, it still built with the production env vars.

**Reply by MilosPaunovic** [reply]
Where and how are you checking which value is it currently?

**Reply by ghost** [reply]
In the compiled dist/spa/app.*.js, looking for my env values there to see if I get dev or prod values. So far only prod.

**Reply by MilosPaunovic** [reply]
I don't think you're supposed to see `NODE_ENV` variable value in there. Try changing the value of `NODE_ENV` for `quasar dev` command, and do a `console.log` somewhere inside your code to make sure.

**Reply by ghost** [reply]
`cross-env NODE_ENV=production quasar dev` shows dev env when I expect prod.
It might be worth noting that I'm using the @quasar/dotenv extension.

**Reply by MilosPaunovic** [reply]
There is a possibility that Quasar overwrites that variable with its own `dev` or `build` commands. Could you create your own variable called, let's say, `ENVIRONMENT` and do logic according to it?

**Reply by ghost** [reply]
Tried `cross-env SOMETHING=bla quasar dev` and `console.log(process.env.SOMETHING)` logged undefined.
Does it work for you?

**Reply by MilosPaunovic** [reply]
I meant to add it the way you add other environment variables to Quasar, as described here:
https://quasar.dev/quasar-cli/handling-process-env

**Reply by ghost** [reply]
Yeah that works. Not sure I understand, how does it help?

**Reply by MilosPaunovic** [reply]
What exactly don't you understand?

**Reply by ghost** [reply]
I'm able to pass build.env to quasar.conf.js and I'm able to use dotenv as
well.
However I need to define the env in the terminal (and eventually in
package.json), so these options don't really help me.

**Comment by hawkeye64** [comment]
There's an easy solution here:
```
quasar build --debug
```

https://quasar.dev/quasar-cli/handling-process-env#values-supplied-by-quasar-cli

**Reply by ghost** [reply]
Still builds with prod env vars (though process.env.DEBUGGING might be true, didn't check).
